src/app/clusters/network-commissioning/network-commissioning.cpp

```diff
@@ -1202,11 +1202,8 @@ void Instance::OnFailSafeTimerExpired()
     mpWirelessDriver->RevertConfiguration();
     mAsyncCommandHandle.Release();
 
-    // Reset state on failsafe expiry.
+    // Mark the network list changed since `mpWirelessDriver->RevertConfiguration()` may have updated it.
     ReportNetworksListChanged();
-    SetLastNetworkId(ByteSpan{});
-    SetLastConnectErrorValue(NullNullable);
-    SetLastNetworkingStatusValue(NullNullable);
```

Comment:
Hmm.  I guess this is maybe trying to handle the case when we just went to no networks and hence _do_ need to reset to null?  Do we have a way of detecting that sanely?

Reply:
Fixed.